route-rs-runtime/src/link/primitive/classify_link.rs

```diff
@@ -104,8 +104,20 @@ impl<C: Classifier + Send + 'static> LinkBuilder<C::Packet, C::Packet> for Class
             let mut task_parks: Vec<Arc<AtomicCell<TaskParkState>>> = Vec::new();
 
             for _ in 0..self.num_egressors.unwrap() {
-                let (to_egressor, from_ingressor) =
-                    crossbeam_channel::bounded::<Option<C::Packet>>(self.queue_capacity);
+                let to_egressor;
+                let from_ingressor;
+                match self.queue_capacity {
+                    None => {
+                        let (s, r) = crossbeam_channel::unbounded::<Option<C::Packet>>();
+                        to_egressor = s;
+                        from_ingressor = r;
+                    }
+                    Some(capacity) => {
+                        let (s, r) = crossbeam_channel::bounded::<Option<C::Packet>>(capacity);
+                        to_egressor = s;
+                        from_ingressor = r;
+                    }
+                }
```

Comment:
I think this block can be more concisely expressed as:
```
let (to_egressor, from_ingressor) = match self.queue_capacity {
  None => crossbeam_channel::unbounded::<Option<C::Packet>>(),
  Some(capacity) => crossbeam_channel::bounded::<Option<C::Packet>>(capacity)
};
```

Reply:
/facepalm Yeah, good point.

Reply:
Since this section seems to be repeated for each link, maybe we can make a runtime utility function for `internalLinkChannel(capacity: Option<usize>)` or something that each link can share.

Reply:
Hmm, I'm not sure I agree, it is only 3 lines of code, and the function will have to be generic over the type we pass around inside the link, might be more trouble that its worth for four locations.